src/units.rs

```diff
@@ -377,6 +377,14 @@ pub fn to_ideal_unit(number: Number) -> Number {
     } else {
       return Number::new(value, Millimeter)
     }
+  } else if number.unit.category() == Time {
+    if number.unit == Hour && value >= d128!(60000000000) && value < d128!(3600000000000) {
+      return Number::new(value/Minute.weight(), Minute)
+    } else if (number.unit == Hour || number.unit == Minute) && value >= d128!(1000000000) && value < d128!(60000000000) {
+      return Number::new(value/Second.weight(), Second)
+    } else if (number.unit == Hour || number.unit == Minute || number.unit == Second) && value >= d128!(1000000) && value < d128!(1000000000) {
+      return Number::new(value/Millisecond.weight(), Millisecond)
```

Comment:
Is it necessary to check the unit here, or could it be set up like the other if clauses?

Reply:
I think it's a very difficult problem to solve uniformly for all units of time. For example, if the result is `0.47 months`, should you convert the result to `14.1 days`, or `2.014285714 weeks`? What happens if the user is expecting the month to be January or February instead of a 30 day month? You'll have similar problems with leap years and leap seconds.

The point is that you'll end up having to have opinions somewhere, and that may not be desirable as you risk presenting the user with information that isn't technically correct for the use case they have in mind.

I chose to implement these checks specifically for these units because it's unambiguous.

Reply:
`to_ideal_unit()` isn't called if the result is `0.47 months`. If you've done a calculation, getting a result in months probably shouldn't happen unless you have some specific calculation where it makes sense.

Here's my suggestion:
```rust
>= 1 year => year // not sure if 1 year is the right number
>= 30 day => day // not sure if 30 days (720 hours) is the right number
>= 1 hour => hour
>= 1 minute => minute
>= 1 second => second
>= 1 millisec => millisec
>= 1 microsec => microsec
else => nanosec
```

It could make sense to return `day` in some cases